src/composable_kernel/composable_kernel/include/utility/type.hpp

```diff
@@ -4,6 +4,84 @@
 #include "integral_constant.hpp"
 #include "enable_if.hpp"
 
+#ifdef __HIPCC_RTC__
+#ifdef WORKAROUND_ISSUE_HIPRTC_TRUE_TYPE
+/// We need <type_traits> for std::remove_reference and std::remove_cv.
+/// But <type_traits> also defines std::true_type, per Standard.
+/// However the latter definition conflicts with
+/// /opt/rocm/include/hip/amd_detail/amd_hip_vector_types.h,
+/// which defines std::true_type as well (which is wrong).
+
+namespace std {
```

Comment:
Can we move these to a common file and include that file everywhere instead ?

Reply:
Would this WA potentially causing the numerical changes in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1237#issuecomment-1040692300 ?

Reply:
Let's fix this issue in follow up PRs